benchmarks/mathvista/mathvista.py

```diff
@@ -0,0 +1,160 @@
+import re
+
+from inspect_ai import Task, task
+from inspect_ai._util.content import ContentImage, ContentText
+from inspect_ai._util.pattern import ANSWER_PATTERN_LETTER, ANSWER_PATTERN_LINE
+from inspect_ai.dataset import Sample, hf_dataset
+from inspect_ai.model import ChatMessageUser
+from inspect_ai.scorer import (
+    CORRECT,
+    INCORRECT,
+    Score,
+    Target,
+    accuracy,
+    scorer,
+    stderr,
+)
+from inspect_ai.scorer._pattern import match_first
+from inspect_ai.solver import Generate, Solver, TaskState, solver
+from inspect_ai.solver._multiple_choice import prompt
+
+SINGLE_ANSWER_TEMPLATE = r"""
+Answer the following multiple choice question. The entire content of your response should be of the following format: 'ANSWER: $LETTER' (without quotes) where LETTER is one of {letters}.
+
+{question}
+
+{choices}
+""".strip()
+
+FREEFORM_TEMPLATE = r"""
+Answer the following question. The entire content of your response should be of the following format: 'ANSWER: $ANSWER' (without quotes) where $ANSWER is one your answer.
+
+{question}
+"""
+
+
+@task
+def MathVistaMultipleChoice() -> Task:
+    dataset = hf_dataset(
+        path="AI4Math/MathVista",
+        split="testmini",
+        sample_fields=record_to_sample,
+        trust=True,
+        shuffle=True,
+    )
+
+    return Task(
+        dataset=dataset,
+        plan=[multi_choice_or_free_form_solver()],
+        scorer=multi_choice_or_free_form_scorer(),
+    )
+
+
+def get_multi_choice_as_letter(record: dict) -> chr:
```

Comment:
Move this function below the `record_to_sample` function (we tend to put helper below so people can read the program from top down)